java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -284,27 +272,18 @@ public WildcardTreeImpl newWildcardTypeArguments(JavaTree extendsOrSuperToken, O
       type);
   }
 
-  public TypeParameterListTreeImpl newTypeParameterList(InternalSyntaxToken openBracketToken, TypeParameterTreeImpl typeParameter, Optional<List<AstNode>> rests,
-                                                        InternalSyntaxToken closeBracketToken) {
+  public TypeParameterListTreeImpl newTypeParameterList(InternalSyntaxToken openBracketToken, TypeParameterTreeImpl typeParameter, Optional<List<Tuple<InternalSyntaxToken,
+      TypeParameterTreeImpl>>> rests, InternalSyntaxToken closeBracketToken) {
     ImmutableList.Builder<TypeParameterTree> typeParameters = ImmutableList.builder();
-    List<AstNode> children = Lists.newArrayList();
-
     typeParameters.add(typeParameter);
-    children.add(typeParameter);
 
     if (rests.isPresent()) {
-      for (AstNode rest : rests.get()) {
-        for (AstNode child : rest.getChildren()) {
-          if (!child.is(JavaPunctuator.COMMA)) {
-            typeParameters.add((TypeParameterTreeImpl) child);
-          }
-
-          children.add(child);
-        }
+      for (Tuple<InternalSyntaxToken, TypeParameterTreeImpl> rest : rests.get()) {
+        //FIXME SONARJAVA-547 commas should be handled rest.first()
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)